### ISSUE TITLE
Improvement: Add maxLength property to ComboBox component

### ISSUE THREAD

**Issue by ikos23** [telerik/kendo-react#914]
## I'm submitting a suggestion (improvement)

* Suggestion for improvement

## Current behavior
`ComboBox` component has `allowCustom` prop which allows us to enter value, but there is no out-of-the-box way to limit what user can enter. And this is quite common scenario.

## Expected behavior
`maxLength` property to be available, so user can control how many chars can be entered (same behavior as for regular input).

## What is the motivation or use case for changing the behavior?
`maxLength` property should be available for  "input-based" component such as ComboBox as there it allows to enter values manually.

## Environment
Any

Browser:
Any

System:
Any

**Comment by nstoychev** [comment]
Thank you for the suggestion @ikos23,

The maxLength functionality can be achieved using the [`filter`](https://www.telerik.com/kendo-react-ui/components/dropdowns/api/ComboBoxProps/#toc-filter) and [`onFilterChange`](https://www.telerik.com/kendo-react-ui/components/dropdowns/api/ComboBoxProps/#toc-onfilterchange) props - https://stackblitz.com/edit/react-7bjust?file=app/main.jsx.

I will add this example to our demos. It is not intuitive at all that the filtering functionality can be used to control the input value. I believe everyone can then modify it to cover similar cases such as minLength or use a regular expression.

**Comment by ikos23** [comment]
Yeah, that is exactly how I did it :) But I thought that something like `maxLength`  could be a much easier (shorter) way to achieve this. 

The "filter" (search bar) is just an `<input />` so the `maxLength` looks kinda logical there.
